plugins/pulp_rpm/plugins/distributors/iso_distributor/distributor.py

```diff
@@ -61,18 +61,19 @@ def publish_repo(self, transfer_repo, publish_conduit, config):
         repo = transfer_repo.repo_obj
         return super(ISODistributor, self).publish_repo(repo, publish_conduit, config)
 
-    def unpublish_repo(self, repo, config):
+    def unpublish_repo(self, transfer_repo, config):
```

Comment:
It is messy, but I think using `transfer_repo` is important in this case because this file mixes repos and repo objects all over the place. There are probably other similar issues elsewhere. I also chose `transfer_repo` because this is what the yum distributor does as well. I would prefer to change the File Distributor's `repo` to `transfer_repo`  Would that work for you?